mlir/include/mlir/Dialect/Linalg/Utils/Utils.h

```diff
@@ -369,7 +369,9 @@ struct GenerateLoopNest {
 /// Returns an attribute list that excludes pre-defined attributes.
 template <typename OpTy>
 SmallVector<NamedAttribute> getPrunedAttributeList(OpTy op) {
-  auto elidedAttrs = llvm::to_vector(op.getAttributeNames());
+  // op.getAttributeNames() doesn't work when the op is linalg::LinalgOp.
```

Comment:
What do you mean by does not work? Can you elaborate or add a test/example to the PR...? Should we just have a template partial specialization for LinalgOp?

Reply:
`LinalgOp` is an interface op... so it doesnt have that method. All `Linalg` ops have the same attributes as a `linalg.generic`. So this works. We could do partial specialization of `LinalgOp`. That would be cleaner. cc @yzhang93

Reply:
yeah, that's also my guess, but I'm not sure. Please also add it to the comment.. Adding a partial specialization makes sense to me.